### PR TITLE
Feature/ccclxxxvi 15

### PR DESCRIPTION
**Pull request by jelysaw**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=993)

## Code reviewers

- [ ] @Andriy97091
- [ ] @irynazaiets
- [ ] @jelysaw
- [ ] @MaksymGromyk
- [ ] @shado1997
- [ ] @tavor118
- [ ] @VolodymyrBodnar
- [ ] @wm139

### Second Level Review

- [ ] @lhalam
- [ ] @alutsysh

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: moneta/src/python/core/utils.py [hunk 1]
```diff
@@ -5,13 +5,12 @@
...
 import string
 from datetime import datetime
-
 from django.contrib.auth.hashers import make_password
 from django.core.mail import send_mail
-
 from settings.settings import DATABASES  # pylint:disable = no-name-in-module, import-error
 
 
+
```
Comment (alutsysh) [inline moneta/src/python/core/utils.py:16]: should be 2 empty lines

File: moneta/src/python/db/currencies.py [hunk 2]
```diff
@@ -59,6 +59,8 @@ def get_currency_rates():
             if item[CURRENCY_CODE_IN_JSON] in currency_list:
                 currency_rates[item[CURRENCY_CODE_IN_JSON]] = (item['rate'])
 
+        print("c", currency_rates)
```
Comment (alutsysh) [inline moneta/src/python/db/currencies.py]: no prints in code

**Review by VolodymyrBodnar — DISMISSED** [review]

File: moneta/src/python/db/income.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
+"""
+Module for interaction with income table in a database.
+"""
+from MySQLdb._exceptions import IntegrityError
```
Comment (alutsysh) [inline moneta/src/python/db/income.py]: add an empty line before

File: moneta/src/python/db/income.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+from core import decorators # pylint:disable = import-error, no-name-in-module
+from core.db import pool_manager as db # pylint:disable = import-error, no-name-in-module
+from core.db.db_helper import DbHelper
+
```
Comment (alutsysh) [inline moneta/src/python/db/income.py:8]: should be 2 empty lines

File: moneta/src/python/db/income.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+class Income(DbHelper):
+    """
+    Class for interacting with income table in a database.
```
Comment (alutsysh) [inline moneta/src/python/db/income.py]: Income

File: moneta/src/python/db/income.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                 amount - amount of edited income, image_id - image for income
+        :return: True if success, else False
+        """
+        sql = f"""
+                UPDATE income
+                SET name=%s, amount=%s, image_id = %s
+                WHERE income.id=%s;
+                """
+        args = (name, amount, image_id, income_id)
+        try:
```
Comment (alutsysh) [inline moneta/src/python/db/income.py]: why try? You have a retry decorator

File: moneta/src/python/db/reset_password.py [hunk 7]
```diff
@@ -27,13 +27,12 @@ def save_password_in_db(user_email, new_password):
         password = new_password
         hashed_password = utils.hash_password(new_password)
         query = f"UPDATE auth_user SET password = '{hashed_password}'" \
-                f" WHERE email = '{user_email}'"
+        f" WHERE email = '{user_email}'"
```
Comment (alutsysh) [inline moneta/src/python/db/reset_password.py]: move to the right as it was before

File: moneta/www/templates/income/edit_income.html [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+                    data:{income_id: id,
+                        csrfmiddlewaretoken: '{{ csrf_token }}'},
+                    success:function (response) {
+                        console.info(response);
+                    },
+                    error(error){
+                        console.error(error);
+                    }
+                });
+            location.href = "http://127.0.0.1:8000/income/"+id+"/delete/";
```
Comment (alutsysh) [inline moneta/www/templates/income/edit_income.html]: do you need full URL?

File: moneta/www/views/income.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    income_user = request.user
+    inc_list = Income.get_income(income_user.id, income_id)
+    icons = StorageIcon.get_icons("income")
+    context = {'income_info': inc_list, "images": icons}
+    if request.POST:
+        income_name = request.POST["name"]
+        income_amount = request.POST["amount"]
+        income_image = request.POST["image_id"]
+        logger = utils.get_logger(__name__)
+        logger.info(str(income_name, income_amount, income_image))
```
Comment (alutsysh) [inline moneta/www/views/income.py]: No text logical info in log message?

File: moneta/src/python/db/currencies.py (hunk 2 above)
Comment (jelysaw) [reply moneta/src/python/db/currencies.py]: It's not mine, but I'll remove it.

File: moneta/www/templates/income/edit_income.html (hunk 8 above)
Comment (jelysaw) [reply moneta/www/templates/income/edit_income.html]: Changed

**Review by alutsysh — APPROVED** [review]

**Review by tavor118 — APPROVED** [review]

**Review by VolodymyrBodnar — APPROVED** [review]

**Review by MaksymGromyk — APPROVED** [review]

**Pull request merged**